### PR TITLE
Add the alphametics exercise

### PR DESCRIPTION
**Pull request by arthurchipdean**
This implements the alphametics exercise. Thanks

### CODE REVIEW

File: exercises/alphametics/example.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+function assignNumbers(numbers, letters, orders) {
+  var output = {};
+  for(var i = 0; i < letters.length; i++) {
+    output[letters[i]] = numbers[orders[i]];
+  }
+  return output;
+}
+
+function getUniqueLetters(string) {
```
Comment (matthewmorgan) [inline exercises/alphametics/example.js]: Interesting!  I have never thought of using `indexOf` this way.  In ES6 we have `Set` now, which cleans up the semantics a little...

File: exercises/alphametics/example.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  }
+  return permute(inputArr);
+}
+
+function getNumberCombinations(set, k) {
+  var i, j, combs, head, tailcombs;
+  if (k > set.length || k <= 0) {
+    return [];
+  }
+  if (k == 1) {
```
Comment (matthewmorgan) [inline exercises/alphametics/example.js]: Let's use `===` here.

File: exercises/alphametics/alphametics.spec.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  it("puzzle with three letters", function() {
+    var puzzle = "I + BB == ILL";
+    var expected = {
+      "I": 1,
+      "B": 9,
+      "L": 0
+    };
+    expect(solve(puzzle)).toEqual(expected);
+  });
+  xit("solution must have unique value for each letter", function() {
```
Comment (matthewmorgan) [inline exercises/alphametics/alphametics.spec.js]: Would you mind putting a blank line between each `it` block please?

File: exercises/alphametics/alphametics.spec.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    var puzzle = "A == B";
+    var expected = null;
+    expect(solve(puzzle)).toEqual(expected);
```
Comment (tejasbubane) [inline exercises/alphametics/alphametics.spec.js]: I think we should make this `.toBeNull()` instead.

File: exercises/alphametics/example.js (hunk 1 above)
Comment (arthurchipdean) [inline exercises/alphametics/example.js]: I agree. I plan on converting this to ES6 soon, there are lots of rooms for optimizations.

**Comment by arthurchipdean** [comment]
@matthewmorgan  I rebased my branch and got rid of the extra commits but it doesn't seem like it actually removed them from the PR. In fact it looks like it duplicated existing ones? What am I doing wrong?

**Comment by matthewmorgan** [comment]
@arthurchipdean I don't usually use `rebase` to squash, it's really more power than you need for this use case.  If you wanted to pick _some_ commits in the history, squash others, then `rebase` is the way to go.  Otherwise, I would just (make sure you're on your branch):

```bash
//this should squash all the commits you made on your branch down to one single one that follows the final commit on master
git reset --soft hash-of-last-commit-on-master
git commit -am 'Add the alphametics exercise'

//then force push back to your branch on GitHub:

git push -f origin master
```

The above should do it.

**Comment by arthurchipdean** [comment]
@matthewmorgan Thank you!

**Comment by arthurchipdean** [comment]
The tests are passing on my pc. I'm not sure what this error is: `gzip: stdin: not in gzip format` Is there something else I need to fix?

**Comment by tejasbubane** [comment]
> The tests are passing on my pc. I'm not sure what this error is: gzip: stdin: not in gzip format Is there something else I need to fix?

@arthurchipdean That was some issue with out config validator. I restarted the build and its green now! 😄

**Comment by matthewmorgan** [comment]
Thanks @arthurchipdean !

**Pull request merged**